govc panic when try to change a vm se-sparse disk to persistent mode

I'm trying to attach an exist vmdk to VM, via govc vm.disk.attach, after attach it, the vm disk mode is always as independent-persistent, then I tried to use vm.disk.change command to change its disk mode, then hit below error:

/ # govc vm.disk.change -vm 0828-RHEL-VM -disk.filePath "[DS-Name] 0828-RHEL-VM/0828-RHEL-VM.vmdk" -mode persistent
panic: interface conversion: types.BaseVirtualDeviceBackingInfo is *types.VirtualDiskSeSparseBackingInfo, not *types.VirtualDiskFlatVer2BackingInfo

goroutine 1 [running]:
github.com/vmware/govmomi/govc/vm/disk.(*change).Run(0xc00018de30, 0x13ae500, 0xc000036048, 0xc00006c7e0, 0x0, 0x0)
        /home/vmware/src/github.com/vmware/govmomi/govc/vm/disk/change.go:149 +0x62f
github.com/vmware/govmomi/govc/cli.Run(0xc000030090, 0x7, 0x7, 0xc000078058)
        /home/vmware/src/github.com/vmware/govmomi/govc/cli/command.go:165 +0x5de
main.main()
        /home/vmware/src/github.com/vmware/govmomi/govc/main.go:99 +0x64
/ # govc vm.disk.change -vm 0828-RHEL-VM -disk.filePath "[ESXST-PromB-FC-50TB-1] 0828-RHEL-VM/0828-RHEL-VM.vmdk" -mode persistent
panic: interface conversion: types.BaseVirtualDeviceBackingInfo is *types.VirtualDiskSeSparseBackingInfo, not *types.VirtualDiskFlatVer2BackingInfo

goroutine 1 [running]:
github.com/vmware/govmomi/govc/vm/disk.(*change).Run(0xc00013de30, 0x13ae500, 0xc0000a0010, 0xc0000867e0, 0x0, 0x0)
        /home/vmware/src/github.com/vmware/govmomi/govc/vm/disk/change.go:149 +0x62f
github.com/vmware/govmomi/govc/cli.Run(0xc0000a8010, 0x7, 0x7, 0xc00008a058)
        /home/vmware/src/github.com/vmware/govmomi/govc/cli/command.go:165 +0x5de
main.main()
        /home/vmware/src/github.com/vmware/govmomi/govc/main.go:99 +0x64

--- a/govc/vm/disk/change.go
+++ b/govc/vm/disk/change.go
@@ -122,6 +122,12 @@ func (cmd *change) CheckDiskProperties(ctx context.Context, name string, disk *t
 	return true
 }
 
+func (cmd *change) setMode(mode *string) {
+	if cmd.mode != "" {
+		*mode = cmd.mode
+	}
+}
+
 func (cmd *change) Run(ctx context.Context, f *flag.FlagSet) error {
 	vm, err := cmd.VirtualMachine()
 	if err != nil {
@@ -146,11 +152,12 @@ func (cmd *change) Run(ctx context.Context, f *flag.FlagSet) error {
 		editdisk.CapacityInKB = int64(cmd.bytes) / 1024
 	}
 
-	backing := editdisk.Backing.(*types.VirtualDiskFlatVer2BackingInfo)
-	backing.Sharing = cmd.sharing
-
-	if len(cmd.mode) != 0 {
-		backing.DiskMode = cmd.mode
+	switch backing := editdisk.Backing.(type) {
+	case *types.VirtualDiskFlatVer2BackingInfo:
+		backing.Sharing = cmd.sharing
+		cmd.setMode(&backing.DiskMode)
+	case *types.VirtualDiskSeSparseBackingInfo:
+		cmd.setMode(&backing.DiskMode)
 	}
 
 	spec := types.VirtualMachineConfigSpec{}

